sql/src/main/java/io/crate/analyze/GenericPropertiesConverter.java

```diff
@@ -96,36 +96,42 @@ static Settings genericPropertiesToSettings(GenericProperties genericProperties,
     static void settingsFromProperties(Settings.Builder builder,
                                        GenericProperties properties,
                                        Row parameters,
-                                       Map<String, Setting> supportedSettings,
+                                       Map<String, Setting<?>> supportedSettings,
                                        boolean setDefaults,
                                        Predicate<String> ignoreProperty,
                                        String invalidMessage) {
         if (setDefaults) {
             setDefaults(builder, supportedSettings);
         }
-        if (!properties.isEmpty()) {
-            for (Map.Entry<String, Expression> entry : properties.properties().entrySet()) {
-                String settingName = entry.getKey();
-                if (ignoreProperty.test(settingName) || ignoreProperty.test(getPossibleGroup(settingName))) {
-                    continue;
-                }
-                SettingHolder settingHolder = getSupportedSetting(supportedSettings, settingName);
-                if (settingHolder == null) {
-                    throw new IllegalArgumentException(String.format(Locale.ENGLISH, invalidMessage, entry.getKey()));
-                }
-                settingHolder.apply(builder, entry.getValue(), parameters);
+        for (Map.Entry<String, Expression> entry : properties.properties().entrySet()) {
+            String settingName = entry.getKey();
+            if (ignoreProperty.test(settingName)) {
+                continue;
+            }
+            String groupName = getPossibleGroup(settingName);
+            if (groupName != null && ignoreProperty.test(groupName)) {
```

Comment:
`containsKey` on a `Map.of(..)` instance throws a NPE if the argument is null, so this change became necessary since I changed some declarations to use `Map.of(..)` instead of a `ImmutableMap`